inc/classes/optimization/CSS/class-combine-google-fonts.php

```diff
@@ -161,19 +165,64 @@ protected function get_combine_tag() {
 		 *
 		 * @param string $display Display value. Can be either auto, block, swap, fallback or optional.
 		 */
-		$display = apply_filters( 'rocket_combined_google_fonts_display', 'swap' );
+		$display        = apply_filters( 'rocket_combined_google_fonts_display', 'swap' );
+		$allowed_values = $this->get_font_display_values();
+		$display        = isset( $allowed_values[ $display ] ) ? $display : 'swap';
+
+		// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedStylesheet
+		return '<link rel="stylesheet" href="' . esc_url( 'https://fonts.googleapis.com/css?family=' . $this->fonts . $this->subsets . '&display=' . $display ) . '" />';
+	}
+
+	/**
+	 * Returns font with display value.
+	 *
+	 * @since  3.5.1
+	 * @author Soponar Cristina
+	 *
+	 * @param  array $font Array containing font tag and matches.
+	 * @return string Google Font tag with display param.
+	 */
+	protected function get_font_with_display( array $font ) {
+		$font_url = html_entity_decode( $font['url'] );
+
+		$query = wp_parse_url( $font_url, PHP_URL_QUERY );
+
+		if ( ! isset( $query ) ) {
+			return $font[0];
+		}
 
-		$allowed_values = [
+		// This filter is documented in inc/classes/optimization/CSS/class-combine-google-fonts.php.
+		$display        = apply_filters( 'rocket_combined_google_fonts_display', 'swap' );
+		$allowed_values = $this->get_font_display_values();
+		$display        = isset( $allowed_values[ $display ] ) ? $display : 'swap';
+
+		$parsed_font = wp_parse_args( $query );
+
+		if ( empty( $parsed_font['display'] ) ) {
+			// Add default display.
+			return str_replace( $font['url'], esc_url( $font_url . '&display=' . $display ), $font[0] );
+		}
+
+		$font_url = str_replace( '&display=' . $parsed_font['display'], '&display=' . $display, $font_url );
+
+		return str_replace( $font['url'], esc_url( $font_url ), $font[0] );
```

Comment:
There's a repeating pattern for the final string replace that can be compressed to reuse the pattern:

```php
$font_url = ! empty( $parsed_font['display'] )
	? str_replace( "&display={$parsed_font['display']}", "&display={$display}", $font_url )
	: "{$font_url}&display={$display}";

return str_replace( $font['url'], esc_url( $font_url ), $font[0] );
```

Also avoid multiple concatenations. Use variable parsing instead by wrapping in double quotes and wrapping vars in curly braces.

Reply:
Resolved in https://github.com/wp-media/wp-rocket/pull/2424/commits/bedb6a8d3dae57c5c61b0cce427e4e555ac008a3#diff-2158ce0f517a79c5a087c9e217f3333aR196